### PR TITLE
Linux: fix read_routes() when an interface has no IP address

### LINKED ISSUE #1567: Route parsing fails after calling `setns`
#### Brief description

When run inside a network namespace with `setns` called form python, scapy fails to read the routes correctly.

#### Environment

- Scapy version: 2.4.0, 2.4.0.dev396
- Python version: 3.7, 3.6
- Operating System: Arch Linux

#### How to reproduce

1. Setup a network namespace
2. Add a inteface to that namespace
3. Run the attaches script

```
sudo ip netns add TEST
sudo ip link set netns TEST dev enp0s25
sudo python netns-crash.py
```

#### Actual result

```
Traceback (most recent call last):
  File "netns-crash.py", line 5, in <module>
    conf.route.resync()
  File "[...]/venv/lib/python3.7/site-packages/scapy/route.py", line 36, in resync
    self.routes = read_routes()
  File "[...]/venv/lib/python3.7/site-packages/scapy/arch/linux.py", line 275, in read_routes
    ifaddr_int = struct.unpack("!I", ifreq[20:24])[0]
UnboundLocalError: local variable 'ifreq' referenced before assignment
```

#### Expected result

The routes should be resynced, then a packet should be sent.
When running the script without the call to `conf.route.resync`, it crashed because it tries to send it on an interface only available in the main namespace

### PR DESCRIPTION
**Pull request by p-l-**
Fixes #1567.

### CODE REVIEW

**Comment by gpotter2** [comment]
We could have some unit tests but this is so unusual that I think we'll be fine without.
To me, ready to be merged

**Comment by guedou** [comment]
@p-l- that would do the trick. Can you add it? Thanks.

```
@mock.patch("scapy.arch.linux.ioctl")
def test_read_routes(mock_ioctl):
    def raise_ioerror(*args, **kwargs):
        raise IOError
    mock_ioctl.side_effect = raise_ioerror
    read_routes()

test_read_routes()
```

**Comment by p-l-** [comment]
i don't understand your test. It crashes (which seems normal to me).

File: scapy/arch/linux.py [hunk 1]
```diff
@@ -269,18 +269,19 @@ def read_routes():
             ifreq = ioctl(s, SIOCGIFADDR, struct.pack("16s16x", iff.encode("utf8")))  # noqa: E501
         except IOError:  # interface is present in routing tables but does not have any assigned IP  # noqa: E501
```
Comment (gpotter2) [inline scapy/arch/linux.py]: He’s trying to trigger this except, by failing at the above `ioctl` @p-l-

**Comment by guedou** [comment]
It should crash on master but works OK with your patch (I did not try it with your fix). It produces the same error as the original issue.

**Comment by p-l-** [comment]
The test was also broken with the patch. Should be fixed now.

**Pull request merged**